Open Judge System/Workers/OJS.Workers.LocalWorker/LocalWorkerService.cs

```diff
@@ -60,53 +49,59 @@ protected override void OnStop()
         {
             logger.Info("LocalWorkerService stopping...");
 
-            foreach (var job in this.jobs)
-            {
-                job.Stop();
-                logger.InfoFormat("{0} stopped.", job.Name);
-            }
+            this.StopJobs(this.jobs);
 
             Thread.Sleep(10000);
 
-            foreach (var thread in this.threads)
+            this.StopThreads(this.threads);
+
+            logger.Info("LocalWorkerService stopped.");
+        }
+
+        private void PopulateJobsAndThreads(
+            ICollection<IJob> jobsToPopulate,
+            ICollection<Thread> threadsToPopulate,
```

Comment:
No need to use 'toPopulate'. Simply `threads`/`jobs` will suffice.

Reply:
it is overriding the private fields `threads` and `jobs`, and makes usage of this. necessary and dangerous if missed.